File: Visualizer/Visualizer/Model/OR.cs
namespace Visualizer.Model
{
    class OR : BitValue
    {
        private BitValue Input1, Input2;

        public OR(BitValue in1, BitValue in2)
        {
            Input1 = in1;
            Input2 = in2;
        }

        public override bool Get()
        {
            return Input1.Get() || Input2.Get();
        }
    }
}

File: Graphula/Graphula/Bits/or.cpp
#include "or.h"

OR::OR(BitValue * in1, BitValue * in2)
{
    _inputs[0] = in1;
    _inputs[1] = in2;
}

BitValue **OR::GetInputs()
{
    return _inputs;
}

int OR::GetInputsCount()
{
    return 2;
}

QString OR::GetName()
{
    return "OR";
}

bool OR::Calc()
{
    return _inputs[0]->Get() || _inputs[1]->Get();
}
